### PR TITLE
[TERRA-483], [TERRA-319] Add workspace create / delete support for AWS

### PR DESCRIPTION
**Pull request by dexamundsen**
Create and delete a workspace (internally creates / handles cloud context) for AWS
Test for the above functionality, run only if server supports AWS platform

```
terra-cli (dexamundsen/terra-483) >> terra server set --name=verily-devel
Terra server is set to verily-devel (CHANGED).

terra-cli (dexamundsen/terra-483) >> terra workspace create --platform=AWS --id=dex-devel-ws-0406-3
Workspace successfully created.
ID:                dex-devel-ws-0406-3
Name:              null
Description:       null
Cloud Platform:    AWS
AWS account:       ***
Cloud console:     https://***.signin.aws.amazon.com/console/
Properties:
Created:           2023-04-07
Last updated:      2023-04-07
# Resources:       0

# Workspace create on WSM via swagger () is included in list
terra-cli (dexamundsen/terra-483) >> terra workspace list
   ID                                        NAME                            GOOGLE PROJECT                  AWS_ACCOUNT      DESCRIPTION                             
...                              
   dex-devel-ws-0406-1                       dex-devel-ws-0406-1             (unset)                         ***              (unset)                                 
 ✓ dex-devel-ws-0406-3                       (unset)                         (unset)                         ***              (unset)                                 
...                      

terra-cli (dexamundsen/terra-483) >> terra workspace describe --workspace=dex-devel-ws-0406-1 
ID:                dex-devel-ws-0406-1
Name:              dex-devel-ws-0406-1
Description:       null
Cloud Platform:    AWS
AWS account:       ***
Cloud console:     https://***.signin.aws.amazon.com/console/
Properties:
Created:           2023-04-07
Last updated:      2023-04-07
# Resources:       0

terra-cli (dexamundsen/terra-483) >> terra workspace describe 
ID:                dex-devel-ws-0406-3
Name:              null
Description:       null
Cloud Platform:    AWS
AWS account:       ***
Cloud console:     https://***.signin.aws.amazon.com/console/
Properties:
Created:           2023-04-07
Last updated:      2023-04-07
# Resources:       0

terra-cli (dexamundsen/terra-483) >> terra workspace delete --quiet
ID:                dex-devel-ws-0406-3
...
Workspace successfully deleted.

terra-cli (dexamundsen/terra-483) >> terra workspace describe 
No workspace set.

terra-cli (dexamundsen/terra-483) >> terra workspace describe --workspace=dex-devel-ws-0406-3
[ERROR] Error fetching workspace: Workspace dex-devel-ws-0406-3 not found.
See /Users/dexamundsen/.terra/logs/terra.log for more information
```

Test run - 

```
terra-cli (dexamundsen/terra-483) >> ./gradlew runTestsWithTag -PtestTag=unit -Pserver=verily-devel -PtestConfig=verily --tests unit.WorkspaceAws
...

Finished running test statusDescribeListReflectCreate() [unit.WorkspaceAws] with result: SUCCESS
```

### CODE REVIEW

File: src/main/java/bio/terra/cli/serialization/userfacing/UFWorkspaceLight.java [hunk 1]
```diff
@@ -111,12 +122,17 @@ public void print() {
...
     OUT.println("Cloud Platform:    " + cloudPlatform);
+
     if (cloudPlatform == CloudPlatform.GCP) {
       OUT.println("Google project:    " + googleProjectId);
       OUT.println(
           "Cloud console:     https://console.cloud.google.com/home/dashboard?project="
               + googleProjectId);
+    } else if (cloudPlatform == CloudPlatform.AWS) {
+      OUT.println("AWS account:       " + awsAccountId);
+      OUT.println("Cloud console:     https://" + awsAccountId + ".signin.aws.amazon.com/console/");
```
Comment (jmczerk) [inline src/main/java/bio/terra/cli/serialization/userfacing/UFWorkspaceLight.java:133]: We should omit this, the user will never have a username/password that will work with this link, instead they will get signed links for console access to individual controlled resources.

File: src/main/java/bio/terra/cli/businessobject/Workspace.java [hunk 2]
```diff
@@ -39,10 +39,11 @@
  */
 public class Workspace {
   private static final Logger logger = LoggerFactory.getLogger(Workspace.class);
-  private UUID uuid;
-  private String userFacingId;
+  private final UUID uuid;
+  private final String userFacingId;
   private CloudPlatform cloudPlatform;
   private String googleProjectId;
+  private String awsAccountId;
```
Comment (jmczerk) [inline src/main/java/bio/terra/cli/businessobject/Workspace.java:46]: We might as well include all the fields from the `AwsCloudContext` here.

File: src/main/java/bio/terra/cli/command/workspace/List.java [hunk 3]
```diff
@@ -72,6 +72,7 @@ private enum Columns implements ColumnDefinition<UFWorkspaceLight> {
     ID("ID", w -> w.id, 40, LEFT),
     NAME("NAME", w -> w.name, 30, LEFT),
     GOOGLE_PROJECT("GOOGLE PROJECT", w -> w.googleProjectId, 30, LEFT),
+    AWS_ACCOUNT("AWS_ACCOUNT", w -> w.awsAccountId, 15, LEFT),
```
Comment (jmczerk) [inline src/main/java/bio/terra/cli/command/workspace/List.java:75]: `environmentAlias` might be more meaningful here (currently `saas` for our `devel` AWS Environent), or maybe a combination of `tenantAlias` + `environmentAlias` (`devel:saas` for `devel`)?

WDYT?

**Review by jmczerk — APPROVED** [review]
Some minor comments on outputs, but LGTM

File: src/main/java/bio/terra/cli/businessobject/Workspace.java (hunk 2 above)
Comment (dexamundsen) [reply src/main/java/bio/terra/cli/businessobject/Workspace.java:46]: No concern with that. At this time did not have a need for more fields. so skipped.  
Keeping in mind storage / other resources, those field might help, at least for logging. 
Adding all the fields to the object, limiting user display fields

File: src/main/java/bio/terra/cli/serialization/userfacing/UFWorkspaceLight.java (hunk 1 above)
Comment (dexamundsen) [reply src/main/java/bio/terra/cli/serialization/userfacing/UFWorkspaceLight.java:133]: Makes sense, removing console link, and displaying only account Id + tenant for now.

File: src/main/java/bio/terra/cli/command/workspace/List.java (hunk 3 above)
Comment (dexamundsen) [reply src/main/java/bio/terra/cli/command/workspace/List.java:75]: AccountId  - likley to know which account their workspace is in.
EnvironmentAlias, tenantAlias - any use?
MajorVersion, OrganizationId - unlikely to be useful

**Review by jwtracy — APPROVED** [review]

**Pull request merged**